### PR TITLE
Empty lobby

### PR DESCRIPTION
**Pull request by Rxann**
Allows the ability to kick everyone from the lobby with one click. Also swaps the bail kick for end kick when kicking chat spammers.

### CODE REVIEW

**Review by Yimura — CHANGES_REQUESTED** [review]
This implementation is pretty stupid, I'd just send a desync to the host and that would empty the lobby as well without interrupting everyone in the lobby and breaking them apart.

Not sure if desync still works like that though.

**Comment by Rxann** [comment]
> This implementation is pretty stupid, I'd just send a desync to the host and that would empty the lobby as well without interrupting everyone in the lobby and breaking them apart.
> 
> Not sure if desync still works like that though.

Are you sure you can desync the host? I thought in the tooltip it said "does not work on host".

**Comment by Yimura** [comment]
> Are you sure you can desync the host? I thought in the tooltip it said "does not work on host".

Yes, exactly because it will backfire when used on the host.
This feature is asking for an empty session, by sending a desync to the host you will be dropped out of the lobby with everyone gone. In the end the same will be achieved without having to kick everyone from the lobby and disturbing their gameplay.

**Comment by Rxann** [comment]
> > Are you sure you can desync the host? I thought in the tooltip it said "does not work on host".
> 
> Yes, exactly because it will backfire when used on the host. This feature is asking for an empty session, by sending a desync to the host you will be dropped out of the lobby with everyone gone. In the end the same will be achieved without having to kick everyone from the lobby and disturbing their gameplay.

Okay, I will reimplement. Should I continue to iterate the g_player_service and check if they are the host, or should I use another way?

**Comment by maybegreat48** [comment]
You can just use the connection manager to close all connections, or the player manager to remove all players from the session

**Comment by Yimura** [comment]
> Okay, I will reimplement. Should I continue to iterate the g_player_service and check if they are the host, or should I use another way?

Check YimMenu what methods there are on if there's a better way to do it.
Being aware of what you can use in the code base is also part of being a competent coder.

**Comment by Yimura** [comment]
> You can use use the connection manager to close all connections, or the player manager to remove all players from the session

That would be even better most likely.

**Comment by gir489returns** [comment]
> This implementation is pretty stupid.

Another LitteTimmy™ stupid shit productions attempt to fit in. It's like he just makes random shit with whatever's exposed in the base to just make PRs. It's fucking annoying. Go back to the sandbox, Timmy.

**Comment by ghost** [comment]
> > This implementation is pretty stupid.
> 
> Another LitteTimmy™ stupid shit productions attempt to fit in. It's like he just makes random shit with whatever's exposed in the base to just make PRs. It's fucking annoying. Go back to the sandbox, Timmy.

im gonna say it to you 🗣️ 🔥

File: src/backend/commands/session/wipe_session.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+	class empty_session : command
+	{
+		using command::command;
+		virtual void execute(const std::vector<uint64_t>&, const std::shared_ptr<command_context> ctx)
+		{
+			g_player_service->iterate([](const player_entry& player) {
+				auto mgr = *g_pointers->m_gta.m_network_player_mgr;
+				mgr->RemovePlayer(player.second->get_net_game_player());
+				g_notification_service->push_success("Empty Session", "Completed");
```
Comment (maybegreat48) [inline src/backend/commands/session/wipe_session.cpp]: This should be outside the iterator callback

**Comment by Rxann** [comment]
the notification has been moved outside the iterator callback

**Review by Yimura — APPROVED** [review]

**Pull request merged**